Dockerfile-cni-plugin

```diff
@@ -16,9 +12,14 @@ ARG TARGETARCH
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=$TARGETARCH GO111MODULE=on \
     go build -o /go/bin/linkerd-cni -mod=readonly -ldflags "-s -w" -v ./cni-plugin/
 
-##
-## Runtime
-##
+FROM --platform=$BUILDPLATFORM curlimages/curl:7.86.0 as fetch
+WORKDIR /build
+ARG TARGETARCH
+ARG LINKERD_REINITIALIZE_PODS_VERSION=v0.1.0
+# TODO: replace org with linkerd, once we make a first release
+RUN curl -LO https://github.com/alpeb/linkerd2-proxy-init/releases/download/reinitialize-pods%2F${LINKERD_REINITIALIZE_PODS_VERSION}/linkerd-reinitialize-pods-${LINKERD_REINITIALIZE_PODS_VERSION}-${TARGETARCH}.tgz
```

Comment:
I didn't consider a separate image given this will always be used alongside the cni plugin, so I figured I'd include it in the same image. I followed the same pattern as the network-validator, publishing a separate binary, but only when we deem necessary. But yeah, this is an additional manual step. I'll do the build here instead :+1: